apps/prairielearn/src/pages/instructorAssessmentAccessOverrides/instructorAssessmentAccessOverrides.js

```diff
@@ -0,0 +1,132 @@
+// @ts-check
+import * as express from 'express';
+const asyncHandler = require('express-async-handler');
+import * as sqldb from '@prairielearn/postgres';
+import * as error from '@prairielearn/error';
+import { getEnrollmentForUserInCourseInstance } from '../../models/enrollment';
+import { selectUserByUid } from '../../models/user';
+
+const router = express.Router();
+const sql = sqldb.loadSqlEquiv(__filename);
+
+router.get(
+  '/',
+  asyncHandler(async (req, res) => {
+    const result = await sqldb.queryAsync(sql.select_assessment_access_policies, {
+      assessment_id: res.locals.assessment.id,
+      student_uid: req.body.student_uid,
+    });
+    res.locals.policies = result.rows;
+    // console.log(res.locals);
+    res.render(__filename.replace(/\.js$/, '.ejs'), res.locals);
+  }),
+);
+
+router.post(
+  '/',
+  asyncHandler(async (req, res) => {
+    if (req.body.__action === 'add_new_override') {
+      const params = {
+        assessment_id: res.locals.assessment.id,
+        created_by: res.locals.authn_user.user_id,
+        credit: req.body.credit,
+        end_date: new Date(req.body.end_date),
+        group_name: req.body.group_name || null,
+        note: req.body.note || null,
+        start_date: new Date(req.body.start_date),
+        user_id: req.body.student_uid,
+        student_uid: req.body.student_uid,
+        group_id: null,
+      };
+      // First, validate if group belongs to the assessment
+      if (params.group_name) {
```

Comment:
We should be doing some more validation with group/individual work. That is: if `assessment.group_work` is `false`, we should require `body.student_uid` and reject a request with `group_name`, and vice versa when `assessment.group_work` is `true`. Ditto for editing.

Reply:
Done in 1a7be5f57ec0056d41b2c46e833c17305bd43f3f